lnwallet/transactions_test.go

```diff
@@ -0,0 +1,923 @@
+			expiry:   503,
+			preimage: "0303030303030303030303030303030303030303030303030303030303030303",
+		},
+		{
+			incoming: true,
+			amount:   4000000,
+			expiry:   504,
+			preimage: "0404040404040404040404040404040404040404040404040404040404040404",
+		},
+	}
+
+	tc.htlcs = make([]channeldb.HTLC, len(htlcData))
+	for i, htlc := range htlcData {
+		preimage, decodeErr := hex.DecodeString(htlc.preimage)
+		if decodeErr != nil {
+			err = fmt.Errorf("Failed to decode HTLC preimage: %v", decodeErr)
+			return
+		}
+
+		tc.htlcs[i].RHash = sha256.Sum256(preimage)
+		tc.htlcs[i].Amt = htlc.amount
+		tc.htlcs[i].RefundTimeout = htlc.expiry
+		tc.htlcs[i].Incoming = htlc.incoming
+	}
+
+	tc.localCsvDelay = 144
+	tc.fundingAmount = 10000000
+	tc.dustLimit = 546
+	tc.feePerKW = 15000
+
+	return
+}
+
+// createNetParams is used by newTestContext to construct new chain parameters
+// as required by the BOLT 03 spec.
+func (tc *testContext) createNetParams(genesisHashStr string) (*chaincfg.Params, error) {
+	params := chaincfg.RegressionNetParams
+
+	// Ensure regression net genesis block matches the one listed in BOLT spec.
+	expectedGenesisHash, err := chainhash.NewHashFromStr(genesisHashStr)
+	if err != nil {
+		return nil, err
+	}
+	if !params.GenesisHash.IsEqual(expectedGenesisHash) {
+		err = fmt.Errorf("Expected regression net genesis hash to be %s, "+
+			"got %s", expectedGenesisHash, params.GenesisHash)
+		return nil, err
+	}
+
+	return &params, nil
+}
+
+// extractFundingInput returns references to the transaction output of the
+// coinbase transaction which is used to fund the channel in the test vectors.
+func (tc *testContext) extractFundingInput() (*Utxo, *wire.TxOut, error) {
+	expectedTxHashHex := "fd2105607605d2302994ffea703b09f66b6351816ee737a93e42a841ea20bbad"
+	expectedTxHash, err := chainhash.NewHashFromStr(expectedTxHashHex)
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to parse transaction hash: %v", err)
+	}
+	expectedUtxo := Utxo{
+		Value: 5000000000,
+		OutPoint: wire.OutPoint{
+			Hash:  *expectedTxHash,
+			Index: 0,
+		},
+	}
+
+	tx, err := tc.block1.Tx(0)
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to get coinbase transaction from "+
+			"block 1: %v", err)
+	}
+	txout := tx.MsgTx().TxOut[0]
+
+	block1Utxo := Utxo{
+		Value: btcutil.Amount(txout.Value),
+		OutPoint: wire.OutPoint{
+			Hash:  *tx.Hash(),
+			Index: 0,
+		},
+	}
+
+	if block1Utxo != expectedUtxo {
+		return nil, nil, fmt.Errorf("Coinbase transaction output from block 1 "+
+			"does not match expected output: expected %v, got %v",
+			expectedUtxo, block1Utxo)
+	}
+
+	return &block1Utxo, txout, nil
+}
+
+// TestCommitmentAndHTLCTransactions checks the test vectors specified in
+// BOLT 03, Appendix C. This deterministically generates commitment and second
+// level HTLC transactions and checks that they match the expected values.
+func TestCommitmentAndHTLCTransactions(t *testing.T) {
+	tc, err := newTestContext()
```

Comment:
All these tests should be able to be safely run in parallel. 

So we can add this fragment to the top of each test:
```go
t.Parallel()
```